packages/manager/src/features/components/PlansPanel/utils.ts

```diff
@@ -109,6 +111,41 @@ export const getRegionsWithCapability = (
   return arrayToList(withCapability ?? []);
 };
 
+interface PlanSoldOutStatus {
+  plan: PlanSelectionType;
+  regionAvailabilities: RegionAvailability[] | undefined;
+  selectedRegionId: Region['id'] | undefined;
+}
+
+/**
+ * Utility to map Premium & GPU plans to a selected region's availability.
+ */
+export const getPlanSoldOutStatus = ({
```

Comment:
Optional naming / comment changes. I like using "is" for boolean things

```suggestion
/**
 * Utility to determine if a plan is sold out based on a region's availability.
 */
export const getIsPlanSoldOut = ({
```
```suggestion
/**
 * Utility to map Premium & GPU plans to a selected region's availability.
 */
export const getPlanSoldOutStatus = ({
```

Reply:
Agreed, done as well!